server/app/controllers/obs.py

```diff
@@ -15,13 +18,81 @@ def obs(obs_id: ObservationId = None, limit_and_offset = None):
     with repository.transaction():
         observation = repository.read_observation(obs_id)
 
+        orbit = None
+
         if observation is None:
             abort(404, "Observation not found")
-    
+
+        orbit = observation
+        if observation['tle'] is not None:
+            orbit = parse_tle(observation['tle'][0], observation['tle'][1])
+
         files = repository.read_observation_files(observation["obs_id"],
             **limit_and_offset)
         files_count = repository.count_observation_files(obs_id)
         satellite = repository.read_satellite(observation["sat_id"])
 
+    # Now tweak some observation parameters to make them more human readable
+    observation = human_readable_obs(observation)
+
     return 'obs.html', dict(obs = observation, files=files,
-        sat_name=satellite["sat_name"], item_count=files_count)
+        sat_name=satellite["sat_name"], item_count=files_count, orbit=orbit)
+
+def parse_tle(tle1: str, tle2: str) -> dict:
+    """ Parses orbital data in TLE format and returns a dictionary with printable orbital elements
+        and other parameters."""
+
+    # First, parse the TLE lines. We don't care about the name.
+    t = TLE.from_lines(line1=tle1, line2=tle2, name="whatevah")
```

Comment:
Could we use here satellite name? Will be any difference?

Reply:
It's not used anywhere (yet), but sure. Updated the code.